src/main/kotlin/org/rust/cargo/project/settings/RustfmtProjectSettingsService.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project.settings
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.*
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import org.jetbrains.annotations.TestOnly
+import org.rust.cargo.project.settings.RustfmtProjectSettingsService.RustfmtState
+import org.rust.cargo.toolchain.RustChannel
+
+val Project.rustfmtSettings: RustfmtProjectSettingsService get() = service()
+
+private const val serviceName: String = "RustfmtProjectSettings"
+
+@State(name = serviceName, storages = [Storage(StoragePathMacros.WORKSPACE_FILE)])
+class RustfmtProjectSettingsService(
+    private val project: Project
+) : SimplePersistentStateComponent<RustfmtState>(RustfmtState()) {
+
+    override fun noStateLoaded() {
+        val oldState = project.rustSettings.settingsState
+        state.useRustfmt = oldState.useRustfmt
+        state.runRustfmtOnSave = oldState.runRustfmtOnSave
```

Comment:
Looks like we need to "restore defaults" in `rustSettings` herel. Otherwise, it will produce quite funny bug. Imagine:
- you set `Use rustfmt instead of the built-in formatter` option to true in current stable plugin
- after it you update the plugin to version with your changes
- and unset the option
- and reopen the project

In this case, I suppose, that `Use rustfmt instead of the built-in formatter` will be unchecked after project reopening but actually, the option will be set again because if service has default state, the state is not persisted to the disk and `noStateLoaded` will be called again. And it will restore old settings again